pylint/extensions/kwargs.py

```diff
@@ -0,0 +1,93 @@
+# Licensed under the GPL: https://www.gnu.org/licenses/old-licenses/gpl-2.0.html
+# For details: https://github.com/PyCQA/pylint/blob/main/LICENSE
+# Copyright (c) https://github.com/PyCQA/pylint/blob/main/CONTRIBUTORS.txt
+
+from __future__ import annotations
+
+from typing import TYPE_CHECKING
+
+from astroid import BoundMethod, UnboundMethod, nodes
+
+from pylint.checkers import BaseChecker
+from pylint.checkers.utils import safe_infer
+from pylint.interfaces import INFERENCE
+
+if TYPE_CHECKING:
+    from pylint.lint import PyLinter
+
+
+class KeywordChecker(BaseChecker):
+    """Checks related to keyword arguments."""
+
+    name = "kwargs"
+    msgs = {
+        "W3501": (
+            "Call to `%s` missing keyword argument `%s`.",
+            "consider-using-keyword-argument",
+            "When using a literal directly in a function call, it's very hard to know what the argument is"
+            " if a positional argument is used because there's no variable name or attribute name to "
+            "rely on when reading. Using a keyword argument there's at least the attribute name to "
+            "help the reader understanding the call.",
+        ),
+    }
+
+    def visit_call(self, node: nodes.Call) -> None:
+        """Check that called functions/methods are provided keyword arguments."""
+        called = safe_infer(node.func)
+
+        if not called:
+            return
+
+        needed_keywords, default_kwarg_names = self._get_args(called)
```

Comment:
This is not type safe. You should do an `isinstance` check before the call to `_get_args`.

Reply:
isinstance check on what exactly? Since _get_args does that already, maybe you can expand on this?

Reply:
`called` is not of the type that you have as annotation in `self._get_args`. Namely `SuccesfulInferenceResult` vs. a number of node types. You either need to expand the annotation in `_get_args` to be broader or you need to check the type of `called` before calling `_get_args`.

Reply:
I don't think you adressed this comment? Did you look at the other comments as well?

Reply:
I believe this has now been addressed